arcgis-ios-sdk-samples/Utility network/Display content of utility network container/DisplayContentUtilityNetworkContainerViewController.swift

```diff
@@ -0,0 +1,366 @@
+        (mapView.map?.operationalLayers as? [AGSLayer])?.forEach { layer in
+            // Make each operational layer visible.
+            layer.isVisible = true
+        }
+        if let previousViewpoint = previousViewpoint {
+            // Return to the viewpoint before container view was entered.
+            mapView.setViewpoint(previousViewpoint) { [weak self] _ in
+                // Enable interaction on the map view.
+                self?.mapView.isUserInteractionEnabled = true
+            }
+        }
+    }
+    
+    /// Create a map.
+    ///
+    /// - Returns: An `AGSMap` object.
+    func makeMap() {
+        let webMapURL = URL(string: "https://sampleserver7.arcgisonline.com/portal/home/item.html?id=813eda749a9444e4a9d833a4db19e1c8")!
+        let map = AGSMap(url: webMapURL)
+        mapView.map = map
+    }
+    
+    /// Create and load the utility network using the feature service URL.
+    func loadUtilityNetwork() {
+        guard let map = mapView.map else { return }
+        utilityNetwork = AGSUtilityNetwork(url: featureServiceURL, map: map)
+        utilityNetwork?.load { [weak self] error in
+            if let error = error {
+                self?.presentAlert(error: error)
+            } else {
+                // Add self as the touch delegate for the map view.
+                self?.mapView.touchDelegate = self
+            }
+        }
+    }
+    
+    /// Get the legend information provided by the feature layers used in the utility network.
+    func fetchLegendInfo() {
+        // Create feature tables from URLs.
+        let electricDistributionTable = AGSServiceFeatureTable(url: featureServiceURL.appendingPathComponent("1"))
+        let structureJunctionTable = AGSServiceFeatureTable(url: featureServiceURL.appendingPathComponent("5"))
+        let legendGroup = DispatchGroup()
+        // Create feature layers using the feature tables.
+        featureLayers = [electricDistributionTable, structureJunctionTable].map(AGSFeatureLayer.init)
+        featureLayers.forEach { layer in
+            legendGroup.enter()
+            // Get the legend information of each layer.
+            layer.fetchLegendInfos { [weak self] (legendInfos, error) in
+                guard let self = self, let legendInfos = legendInfos else { return }
+                defer { legendGroup.leave() }
+                self.legendInfos.append(contentsOf: legendInfos)
+                if let error = error {
+                    self.presentAlert(error: error)
+                }
+            }
+        }
+        legendGroup.notify(queue: .main) { [weak self] in
+            guard let self = self else { return }
+            let featureLayerSymbols = self.legendInfos.reduce(into: [(String, AGSSymbol)]() ) { specifiedSymbols, legendInfo in
+                if !legendInfo.name.isEmpty, let symbol = legendInfo.symbol {
+                    specifiedSymbols.append((legendInfo.name, symbol))
+                }
+            }
```

Comment:
>  but it is also OK to fetch legend info and create symbol swatches all at once (so that we can use [LegendItem] instead of the array of tuples).

What do you mean? I only created the array of tuples so that I could make swatches for the feature layer legend items and the added line symbols all at once. Do you think there's a more efficient solution?

Reply:
I'm suggesting sth (drafted in the editor, might not compile) like 👇 that doesn't need the temp tuple.
```swift
            featureLayers.forEach { layer in
                legendGroup.enter()
                // Get the legend information of each layer.
                layer.fetchLegendInfos { [weak self] legendInfos, _ in
                    defer { legendGroup.leave() }
                    guard let legendInfos = legendInfos else { return }
                    legendInfos.forEach { info in
                        info.symbol.createSwatch { image, _ in
                            defer { getSymbolsGroup.leave() }
                            guard let image = image else { return }
                            legendItems.append(LegendItem(name: info.name, image: image))
                        }
                    }
                }
            }
```

But I like the idea to split into 2 methods when it's too lengthy, so maybe just keep what you have here.

Reply:
Oh I see. Yeah I agree – I like having 2 methods so that I can utilize it for the line symbols as well.